core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -735,7 +757,11 @@ object SparkSubmit {
     }
 
     try {
-      mainMethod.invoke(null, childArgs.toArray)
+      if (isSparkApp) {
+        mainMethod.invoke(null, childArgs.toArray, childSparkConf, childSysProps)
```

Comment:
What is the difference between `childSparkConf` and `childSysProps`?

SparkSubmit currently uses system properties because it has no other way to communicate Spark configs to the apps. But `SparkApp` provides that, so basically, there's no need to tell the child app about system properties, since SparkSubmit shouldn't be setting any when calling a `SparkApp`.

From my previous comments, instead of `childSysProps` here you probably want some way to specify a custom environment (a.k.a. the `env` argument in `SparkLauncher(Map<String, String> env)`). That should not be mixed with system properties or Spark configuration.